R/report-plots.R

```diff
@@ -300,3 +300,400 @@ district_barplot <- function(naomi_output,
+#' @param age_disag Age group filter.
+#' @param sex_disag Sex filter.
+#'
+#' @export
+
+bar_plotly <- function(df,
+                       ind,
+                       quarter,
+                       age_disag = "Y015_049",
+                       sex_disag  = "both") {
+
+  remove_buttons <- c("zoomIn2d", "zoomOut2d", "pan2d", "select2d", "lasso2d",
+                      "autoScale2d", "resetScale2d", "hoverClosestCartesian",
+                      "hoverCompareCartesian", "zoom")
+
+
+  output_source <- paste0("Naomi estimate ", quarter)
+
+  plot_data <- df %>%
+    dplyr::filter(indicator == ind,
+                  calendar_quarter == quarter,
+                  age_group == age_disag,
+                  sex == sex_disag,
+                  !is.na(mean))
+
+  buttons <- dropdown_buttons(plot_data, "area_level_label")
+
+  if (ind == "prevalence") {
+    title <- "HIV prevalence"
+  } else if (ind == "art_coverage") {
+    title <- "ART coverage"
+  }
+  plot_title <- paste("<b>", title, ": ", "Household survey vs. ",
+    output_source, "<b>")
+
+
+  mrg <- list(l = 50, r = 50, b = 50, t = 120, pad = 20)
+
+  plot_data <- plot_data[order(plot_data$source), ]
+
+
+  final_plot <- plotly::plot_ly(
+    data = plot_data,
+    type = "bar",
+    color = ~ as.factor(source),
+    colors = c("#07bbc1", "#f68e1f", "#87c440"),
+    x = ~area_name,
+    y = ~mean,
+    hoverinfo = "text",
+    text = ~paste("</br>", area_name,
+                  "</br>", source,
+                  "</br>", round(mean * 100, 2),
+                  " (", round(upper * 100, 2), "-",
+                  round(lower * 100, 2),  "%)"),
+    error_y = ~list(symmetric = FALSE,
+                    arrayminus = mean - lower,
+                    array = upper - mean,
+                    color = "#000000"),
+    transforms = list(
+                   list(
+                   type = "filter",
+                   target = ~area_level_label,
+                   operation = "=",
+                   value = sort(plot_data$area_level_label)[1]))) %>%
+    plotly::layout(
+      margin = mrg,
+      xaxis = list(type = "category",
+                   categoryarray =  ~source,
+                   categoryorder = "array",
+                   title = list(text = "")),
+      yaxis = list(tickformat = ".0%",
+                   tickmode = "array",
+                   title = list(text = title, font = list(size = 10))),
+      title = list(text = plot_title,
+                   font = list(size = 13, colour = "black", face = "bold")),
+      legend = list(title = list(text = "Data source",
+                                 font = list(size = 10))),
+      updatemenus = list(
+        list(
+          type = "dropdown",
+          y = 1.2,
+          x  = 0.2,
+          active = 0,
+          buttons = buttons
+        )
+      )
+    ) %>%
+    plotly::config(modeBarButtonsToRemove = remove_buttons, displaylogo = FALSE)
+
+  suppressWarnings(final_plot)
```

Comment:
Why are we suppressing the warnings here? What warnings is this producing? It would be better to address the root issue if possible